BridgeEmulator/functions/lightRequest.py

```diff
@@ -224,16 +224,20 @@ def syncWithLights(lights, addresses, users, groups): #update Hue Bridge lights
                         try:
                             light_state = protocol.get_light_state(addresses[light], lights[light])
                             lights[light]["state"].update(light_state)
+                            lights[light]["state"]["reachable"] = True
                         except Exception as e:
                             lights[light]["state"]["reachable"] = False
-                            logging.warning("light not reachable: %s", e)
+                            lights[light]["state"]["on"] = False
+                            logging.warning(lights[light]["name"] + " is unreachable: %s", e)
 
                 if addresses[light]["protocol"] == "native":
                     light_data = json.loads(sendRequest("http://" + addresses[light]["ip"] + "/get?light=" + str(addresses[light]["light_nr"]), "GET", "{}"))
                     lights[light]["state"].update(light_data)
+                    lights[light]["state"]["reachable"] = True
                 elif addresses[light]["protocol"] == "hue":
                     light_data = json.loads(sendRequest("http://" + addresses[light]["ip"] + "/api/" + addresses[light]["username"] + "/lights/" + addresses[light]["light_id"], "GET", "{}"))
                     lights[light]["state"].update(light_data["state"])
+                    lights[light]["state"]["reachable"] = True
```

Comment:
Here is see also an "overwrite" of a status. The Hue bridge says which lights are reachable. This code line says that if the bridge answers the light is reachable. I think this line could be deleted

Reply:
Honestly not sure about other protocols besides the yeelight and tasmota protocols since those are the only ones I use and have looked into. But you may be correct about Hue. It seems like line 239 updates the state from the hue bridge.